core/src/main/java/org/streamprocessor/core/transforms/DeIdentifyFn.java

```diff
@@ -154,10 +161,13 @@ public static Object getFieldToken(String tokenFullRef, Firestore db) {
                 return futureTransaction.get();
             }
         } catch (Exception e) {
-            LOG.info("tokens: something went wrong... race condition?");
-            LOG.error(e.toString());
-            // return null;
-            throw new RuntimeException(e.toString());
+            LOG.error(
+                    "exception[{}] step[{}] details[{}] guess[race condition?]",
+                    e.getClass().getName(),
+                    "DeIdentifyFn.getFieldToken()",
+                    e.toString());
+            throw new RuntimeException(
+                    e.toString()); // DEBATABLE: Why through new and wrapped in a RuntimeException?
```

Comment:
Perhaps we can create a new ticket for this and also investigate why we're catching very broad `Exception` instead of something specific / we could handle specific ones differently

Reply:
I see we're doing this in other places as well, so perhaps we can just look over where we're catching `Exception` in the future and be more specific where we can be

Reply:
Yes I also see this as concerning. But since we don't know how the application behaves just yet we can maybe let the error logs speak for themselves and adjust accordingly as we progress?